### PR TITLE
Define-Discovery-API

### PR DESCRIPTION
**Pull request by IdoJS**

### CODE REVIEW

File: packages/scalecube-discovery/src/Discovery/Discovery.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    let seed = getSeed({ seedAddress });
+    const subjectNotifier = new ReplaySubject<Endpoint[]>(1);
+
+    seed = addToCluster({ seed, address, endPoints, subjectNotifier });
+    notifyAllListeners({ seed });
+
+    return Object.freeze({
+      end: () => {
+        seed = removeFromCluster({ seed, address });
+        subjectNotifier && subjectNotifier.unsubscribe();
```
Comment (dmitriyslaym) [inline packages/scalecube-discovery/src/Discovery/Discovery.ts]: As I understand, you can call **unsubscribe()** on the subscription, not on the stream itself

Comment (IdoJS) [reply packages/scalecube-discovery/src/Discovery/Discovery.ts]: i'll change it to subjectNotifier .complete()

File: package.json [hunk 2]
```diff
@@ -24,7 +24,7 @@
   "scripts": {
     "bootstrap": "lerna bootstrap --npm-client=yarn",
     "build": "lerna run build",
-    "test": "lerna run test",
+    "test": "lerna run test:lint:prettier",
```
[the remark this answers is not in the document]
Comment (IdoJS) [reply package.json]: i'll split it to lint&prettier and test

File: packages/scalecube-microservice/tests/implementations/Microservice.spec.ts [hunk 3]
```diff
@@ -111,4 +117,28 @@ describe('Test the creation of Microservice', () => {
...
+      const greetingService1: Service = {
+        definition: greetingServiceDefinition,
+        reference: new GreetingService(),
+      };
+      const greetingService2: Service = {
+        definition: greetingServiceDefinition2,
+        reference: new GreetingService2(),
+      };
+      expect.assertions(3);
+
```
Comment (idanilt) [inline packages/scalecube-microservice/tests/implementations/Microservice.spec.ts]: asset one the API
Separate it to multiple tests (each test, check one thing)

**Comment by idanilt** [comment]
- Move private from API
- Put JSDoc comments on API
- DiscoveryConnect --> DiscoveryOptions
- DiscoveryCreate / end --> destroy
- createDiscovery(DiscoveryOptions):Discovery
- Think about notifier and events
- Remove cluster from API

File: packages/examples/src/index.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { Microservices } from '@scalecube/scalecube-microservice';
+import Microservices from '@scalecube/scalecube-microservice';
```
Comment (dmitriyslaym) [inline packages/examples/src/index.ts]: { Microservices } is named import

File: packages/scalecube-discovery/src/api/Discovery.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { Observable } from 'rxjs';
+import { Item } from '../helpers/types';
+
+/**
+ * Discovery provide functionality to receive/send information about nodes in the cluster
+ * and to remove Node from the Cluster.
+ *
+ * `destroy: Promise<string>` - remove Microservice Node from the Cluster
```
Comment (dmitriyslaym) [inline packages/scalecube-discovery/src/api/Discovery.ts]: put the comment next to the method

File: packages/scalecube-discovery/src/api/Discovery.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+ *
+ * `notifier: Observable<Endpoint[]>` - notify on endpoints in the Cluster
```
Comment (dmitriyslaym) [inline packages/scalecube-discovery/src/api/Discovery.ts]: put the comment next to the method

File: packages/scalecube-discovery/src/api/Discovery.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import { Observable } from 'rxjs';
+import { Item } from '../helpers/types';
+
+/**
+ * Discovery provide functionality to receive/send information about nodes in the cluster
```
Comment (dmitriyslaym) [inline packages/scalecube-discovery/src/api/Discovery.ts]: provides

File: packages/scalecube-discovery/src/api/DiscoveryOptions.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import { Item } from '../helpers/types';
+
+/**
+ * DiscoveryOptions - The parameters that are required to add Node to the Cluster
```
Comment (dmitriyslaym) [inline packages/scalecube-discovery/src/api/DiscoveryOptions.ts]: seems confusing - this is the parameters to create the discovery

File: packages/scalecube-discovery/src/api/DiscoveryOptions.ts [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+ *
+ * @param address - Microservice unique address.
```
Comment (dmitriyslaym) [inline packages/scalecube-discovery/src/api/DiscoveryOptions.ts]: put comment for each property next to the property itself

Comment (dmitriyslaym) [reply packages/scalecube-discovery/src/api/DiscoveryOptions.ts]: can you provide an example of this address?

File: packages/scalecube-discovery/src/api/DiscoveryOptions.ts [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+ *
+ * @param endPoints - Endpoints to share in the Cluster.
```
Comment (dmitriyslaym) [inline packages/scalecube-discovery/src/api/DiscoveryOptions.ts]: seems better to explain what is endpoint

File: packages/scalecube-discovery/src/api/DiscoveryOptions.ts [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+ *
+ * @param seedAddress - address of the Cluster seed.
```
Comment (dmitriyslaym) [inline packages/scalecube-discovery/src/api/DiscoveryOptions.ts]: provide example please

File: packages/scalecube-discovery/README.md [hunk 12]
```diff
@@ -0,0 +1,28 @@
+[![Join the chat at https://gitter.im/scalecube-js/Lobby](https://badges.gitter.im/scalecube-js/Lobby.svg)](https://gitter.im/scalecube-js/Lobby?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
+
+> **NOTICE** versions 0.0.x are experimental without LTS or the API and behavior might change from patch to patch
```
Comment (idanilt) [inline packages/scalecube-discovery/README.md]: 0.0.x and 0.1.x

**Comment by dmitriyslaym** [comment]
Why don't we have Travis running on this PR?

**Pull request merged**